--- a/m4/libxml2.m4	2025-01-23 10:46:42.916793694 +0100
+++ -	2025-09-26 19:20:48.502502027 +0200
@@ -58,7 +58,7 @@
             AC_REQUIRE([PKG_PROG_PKG_CONFIG])
             m4_ifdef([PKG_PROG_PKG_CONFIG], [PKG_PROG_PKG_CONFIG()], [:])
 
-            if test -x "$PKG_CONFIG"; then
+            if test -n "$PKG_CONFIG"; then
 
                 LIBXML2_CFLAGS="`$PKG_CONFIG --cflags libxml-2.0`"
 
