diff --git a/launcher/MangoHud.cpp b/launcher/MangoHud.cpp
index ab79f418..ab685b82 100644
--- a/launcher/MangoHud.cpp
+++ b/launcher/MangoHud.cpp
@@ -27,13 +27,14 @@
 #include "Json.h"
 #include "MangoHud.h"
 
-#ifdef __GLIBC__
+#ifdef Q_OS_LINUX
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #define UNDEF_GNU_SOURCE
 #endif
 #include <dlfcn.h>
 #include <linux/limits.h>
+#include <link.h>
 #endif
 
 namespace MangoHud {
@@ -119,7 +120,7 @@ QString getLibraryString()
 
 QString findLibrary(QString libName)
 {
-#ifdef __GLIBC__
+#ifdef Q_OS_LINUX
     const char* library = libName.toLocal8Bit().constData();
 
     void* handle = dlopen(library, RTLD_NOW);
@@ -128,14 +129,14 @@ QString findLibrary(QString libName)
         return {};
     }
 
-    char path[PATH_MAX];
-    if (dlinfo(handle, RTLD_DI_ORIGIN, path) == -1) {
+    struct link_map *linkmap;
+    if (dlinfo(handle, RTLD_DI_LINKMAP, &linkmap) == -1) {
         qCritical() << "dlinfo() failed:" << dlerror();
         dlclose(handle);
         return {};
     }
 
-    auto fullPath = FS::PathCombine(QString(path), libName);
+    auto fullPath = QString(linkmap->l_name);
 
     dlclose(handle);
     return fullPath;
