--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -25,13 +25,6 @@
     LINK_LIBRARIES KF5::Libkleo Qt${QT_MAJOR_VERSION}::Test
 )
 
-ecm_add_test(
-    keyresolvercoretest.cpp
-    keyresolvercoretest.qrc
-    TEST_NAME keyresolvercoretest
-    LINK_LIBRARIES KF5::Libkleo Qt${QT_MAJOR_VERSION}::Test
-)
-
 ecm_add_tests(
     editdirectoryservicedialogtest.cpp
     LINK_LIBRARIES KF5::Libkleo KF5::WidgetsAddons Qt${QT_MAJOR_VERSION}::Widgets Qt${QT_MAJOR_VERSION}::Test
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -33,7 +33,6 @@
 ecm_add_tests(
     keyselectioncombotest.cpp
     keyserverconfigtest.cpp
-    newkeyapprovaldialogtest.cpp
     LINK_LIBRARIES KF5::Libkleo Qt${QT_MAJOR_VERSION}::Widgets Qt${QT_MAJOR_VERSION}::Test
 )
